master_server/master_server.py

```diff
@@ -89,107 +89,155 @@ def change_state(request_id, input_file):
       number_of_retries_till_now: number_of_retries_till_now})
-
   DB.session.commit()
 
 @APP.route('/', methods=['GET', 'POST'])
 def hello_world():
-
   return(' You have reached master server \n')
 
 @APP.route('/register', methods=['GET', 'POST'])
 def add_vm_address():
 #  Add a VM to working_vm_address_list list.
-
   address = request.data.decode('utf-8')
   if address not in working_vm_address_list:
     working_vm_address_list.append(address)
   return {'message': 'success'}
 
 @APP.route('/upload')
-
 def upload_file_webpage():
-
 #  Serve web page
   return render_template('upload.html')
 
-@APP.route('/uploader', methods=['GET', 'POST'])
-
+@APP.route('/assign_task', methods=['GET', 'POST'])
 def upload_file():
   """Accept proto file from user and
      assign it to VM.
   """
-
+  task_status_response = Request_pb2.TaskStatusResponse()
   f = request.files['file']
   read = f.read()
   vm_n = find_working_vm()
   if vm_n == 'NOT':
-    return 'try again later'
+    task_status_response.status = Request_pb2.TaskStatusResponse.REJECTED
+#    print('xxxx')
+    if args.debug_mode == 'b':
+      return task_status_response.SerializeToString()
+    else:
+      return str(task_status_response)
   else:
     try:
       working_vm_address_list.remove(vm_n)
       task_request = Request_pb2.TaskRequest()
       task_request.ParseFromString(read)
       requ = create_new_request(
-
           read, 'accepted', vm_n, task_request. number_of_retries, 0)
       task_request.request_id = requ.request_id
-      response = requests.post(url=vm_n, files={'task_request':
+      response = requests.post(url=vm_n + '/assign_task', files={'task_request':
           task_request.SerializeToString()})
-      return 'file uploaded successfully,your request_id is '+ \
-          str(requ.request_id)+'  ' + ' address is ' + str(requ.last_vm_assigned)+ \
-          ' number of retries allowed is ' + str(requ.number_of_retries_allowed)
-    except:
-      return 'try again later'
+      task_status_response.ParseFromString(response.content)
+#      print(task_status_response)
+      t = threading.Thread(target= retry_after_timeout, args= (task_request.request_id, task_request.timeout, task_request.number_of_retries))
+      t.start()
+      if task_status_response.status == Request_pb2.TaskStatusResponse.ACCEPTED:
+#        print('yyyy')
+#        task_status_response.status = Request_pb2.TaskStatusResponse.ACCEPTED
+        if args.debug_mode == 'b':
+          return task_status_response.SerializeToString()
+        else:
+          return str(task_status_response)
+    except Exception as e:
+      task_status_response.status = Request_pb2.TaskStatusResponse.REJECTED
+      print(e)
+      if args.debug_mode == 'b':
+        return task_status_response.SerializeToString()
+      else:
+        return str(task_status_response)
+
+def retry_after_timeout(request_id, timeout, number_of_retries):
+#  print('qqqq '+str(timeout)+'  '+str(request_id))
```

Comment:
remove all the print statements (both commented and non-commented. we should always use logging

Reply:
I have removed all print statements.